File: wedding-repository-frontend/src/features/audit-logs/model/audit-log.ts
export type AuditResultStatus = "SUCCESS" | "FAIL";

export type AuditLog = {
  id: number;
  occurredAt: string; // ISO OffsetDateTime
  actorUserId: number;
  actorUsername: string;
  actorGroupName: string;
  actionCode: string;
  moduleKey: string;
  targetType: string | null;
  targetId: string | null;
  targetLabel: string | null;
  resultStatus: AuditResultStatus;
  description: string | null;
  errorMessage: string | null;
  details: Record<string, unknown> | null;
};

export type AuditLogPage = {
  items: AuditLog[];
  totalElements: number;
  totalPages: number;
  page: number;
  size: number;
};

export type AuditLogSearchParams = {
  from?: string;
  to?: string;
  username?: string;
  actionCode?: string;
  resultStatus?: AuditResultStatus | "";
  page?: number;
  size?: number;
};

export const RESULT_STATUS_CONFIG: Record<AuditResultStatus, { label: string; className: string }> = {
  SUCCESS: { label: "Thành công", className: "bg-green-50 text-green-700" },
  FAIL: { label: "Thất bại", className: "bg-rose-50 text-rose-700" }
};

File: wedding-repository-backend/src/main/java/com/uit/weddingmanagement/modules/audit/domain/model/AuditLog.java
package com.uit.weddingmanagement.modules.audit.domain.model;

import java.time.OffsetDateTime;
import java.util.Map;

/**
 * Model đại diện cho một bản ghi nhật ký hệ thống (Audit Log).
 * Record này chứa thông tin chi tiết về các hành động người dùng thực hiện trên hệ thống.
 */
public record AuditLog(
    Long id,               // ID duy nhất của bản ghi
    OffsetDateTime occurredAt, // Thời điểm xảy ra hành động
    Long actorUserId,      // ID của người thực hiện hành động
    String actorUsername,  // Tên đăng nhập của người thực hiện
    String actorGroupName, // Tên nhóm của người thực hiện (nếu có)
    String actionCode,     // Mã hành động (ví dụ: CREATE_USER, UPDATE_WEDDING)
    String moduleKey,      // Khóa của module liên quan (ví dụ: USER, AUDIT)
    String targetType,     // Loại đối tượng bị tác động
    String targetId,       // ID của đối tượng bị tác động
    String targetLabel,    // Nhãn hiển thị của đối tượng bị tác động
    AuditResultStatus resultStatus, // Trạng thái kết quả của hành động (SUCCESS/FAIL)
    String description,    // Mô tả chi tiết về hành động
    String errorMessage,   // Thông báo lỗi (nếu có)
    Map<String, Object> details // Các thông tin chi tiết bổ sung dưới dạng key-value
) {
}
